packages/ui-voip/src/providers/VoipProvider.tsx

```diff
@@ -39,7 +39,15 @@ const VoipProvider = ({ children }: { children: ReactNode }) => {
 	const dispatchToastMessage = useToastMessageDispatch();
 
 	// Refs
-	const remoteAudioMediaRef = useRef<HTMLAudioElement>(null);
+	const remoteAudioMediaRef = useCallback(
+		(node: HTMLMediaElement | null) => {
+			if (!node || !voipClient) {
+				return;
```

Comment:
Consider removing the old reference for the element on `voipClient`'s internal state to avoid leaks.